pulsar-client-cpp/lib/BatchMessageContainer.cc

```diff
@@ -93,6 +93,10 @@ void BatchMessageContainer::sendMessage() {
     impl_->metadata.set_num_messages_in_batch(messagesContainerListPtr_->size());
     compressPayLoad();
 
+    SharedBuffer encryptedPayload;
```

Comment:
Creating the `SharedBuffer` will need to malloc some stuff, can we just doing when encryption is enabled?

Reply:
You are right. I have allocated a byte to have valid object. Will try to refactor it before merge.

Reply:
Checking again, it seems we are ok to without allocating anything as well.
https://github.com/apache/incubator-pulsar/blob/master/pulsar-client-cpp/lib/SharedBuffer.h#L46
The ptr will be null and will be an issue if we use the buffer at this point. We will error out if encryption fails anyway, so guaranteed to have non null if success. Do you see any issues @merlimat ?